### PR TITLE
Support for config IP when genegrate pki certificate

### PR DESCRIPTION
**Pull request by forgocode**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Add the kubeApiserverCertSANs field to the KwokctlConfiguration to add a SAN when creating the cluster.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/apis/internalversion/kwokctl_configuration_types.go
Comment (wzshiming) [inline pkg/apis/internalversion/kwokctl_configuration_types.go:215]: Refer to the name of this field in Kubeadm

```suggestion
	// KubeApiserverCertSANs sets extra Subject Alternative Names for the API Server signing cert.
	KubeApiserverCertSANs []string
```

https://github.com/kubernetes/kubernetes/blob/b3056ea9bdb4d8d8e0fbb15e08dba1bfa294b813/cmd/kubeadm/app/apis/kubeadm/types.go#L160-L159

File: pkg/kwokctl/runtime/binary/cluster.go
Comment (wzshiming) [inline pkg/kwokctl/runtime/binary/cluster.go:23]:
```suggestion
	"net"
```

File: pkg/kwokctl/cmd/create/cluster/cluster.go
Comment (wzshiming) [inline pkg/kwokctl/cmd/create/cluster/cluster.go:122]: I'd like to reduce the flags parameter, maybe that's enough ?

`kwokctl create cluster --config ./config.yaml`

``` yaml
# ./config.yaml
kind: KwokctlConfiguration
apiVersion: config.kwok.x-k8s.io/v1alpha1
options:
  kubeApiserverCertSANs:
    - 127.0.0.1
    - localhost
    - 10.0.0.1
```

**Review by wzshiming — COMMENTED** [review]
/ok-to-test

File: pkg/kwokctl/cmd/create/cluster/cluster.go
Comment (forgocode) [reply pkg/kwokctl/cmd/create/cluster/cluster.go:122]: ok, i got it! I will make modifications according to your suggestion.
But i think the flag is needed for test engineer even if used infrequently

**Comment by wzshiming** [comment]
Please rebase this branch and don't merge the main into your branch.

**Comment by wzshiming** [comment]
I have refactored this section to get all the IPs that may be accessed when creating the cluster and add them to the credentials, I think this should temporarily alleviate your problem, can you test it #642?

The fields you added in this PR are also necessary in the case of access through the domain.

**Comment by forgocode** [comment]
> I have refactored this section to get all the IPs that may be accessed when creating the cluster and add them to the credentials, I think this should temporarily alleviate your problem, can you test it #642?
> 
> The fields you added in this PR are also necessary in the case of access through the domain.

yes, it works, now i can access to apiserver by physical IP.  #642

**Comment by wzshiming** [comment]
please fix pull-kwok-verify-main https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/636/pull-kwok-verify-main/1667430266894815232

`./hack/update-api-docs.sh`

**Comment by wzshiming** [comment]
Please rebase this branch and don't merge the main into your branch.

File: pkg/apis/config/v1alpha1/kwokctl_configuration_types.go
Comment (wzshiming) [inline pkg/apis/config/v1alpha1/kwokctl_configuration_types.go:328]:
```suggestion
	// KubeApiserverCertSANs sets extra Subject Alternative Names for the API Server signing cert.
```

File: pkg/apis/internalversion/kwokctl_configuration_types.go
Comment (wzshiming) [inline pkg/apis/internalversion/kwokctl_configuration_types.go:213]:
```suggestion
	// KubeApiserverCertSANs sets extra Subject Alternative Names for the API Server signing cert.
```

File: pkg/kwokctl/pki/pki.go
Comment (wzshiming) [inline pkg/kwokctl/pki/pki.go:61]: The contents of the array need to remain unique, not just the two addresses.

**Comment by wzshiming** [comment]
Or do you want to wait until I merge PR #642, then rebase this PR, which should make this PR easier

**Comment by forgocode** [comment]
> Or do you want to wait until I merge PR #642, then rebase this PR, which should make this PR easier
Please follow your plan, git is unfamiliar for me because i always use svn in the past. This is a great chance for me to practice git。

**Comment by forgocode** [comment]
> Or do you want to wait until I merge PR #642, then rebase this PR, which should make this PR easier

Please follow your plan, git is unfamiliar for me because i always use svn in the past. This is a great chance for me to practice git。

**Review by wzshiming — APPROVED** [review]
/approve
/lgtm

Thank you for your contribution

**Comment by forgocode** [comment]
> /approve /lgtm
> 
> Thank you for your contribution

I get so more knowledge about golang project and git, thanks for your guidance. By the way, can you provide a doc for developer about what  should do when he/she want to contribue for this project. Example, how to check code format or if need to generate md files again.

**Comment by wzshiming** [comment]
Actually, it's already there, just not in such detail.
https://kwok.sigs.k8s.io/docs/contributing

**Comment by forgocode** [comment]
> Actually, it's already there, just not as detailed. https://kwok.sigs.k8s.io/docs/contributing

Yes, i had read this page, but maybe it can be more detailed.

**Comment by wzshiming** [comment]
Sure, if I had more time or someone can work on this.

**Pull request merged**
